### PR TITLE
Miscellaneous improvements

### PR DESCRIPTION
**Pull request by RemiCardona**
With fixes for bugs #70 and #80.

### CODE REVIEW

File: aioamqp/__init__.py [hunk 1]
```diff
@@ -58,6 +59,14 @@ def connect(host='localhost', port=None, login='guest', password='guest',
         factory, host, port, **create_connection_kwargs
     )
 
+    # these 2 flags *may* show up in sock.type. They are only available on linux
+    # see https://bugs.python.org/issue21327
+    nonblock = getattr(socket, 'SOCK_NONBLOCK', 0)
+    cloexec = getattr(socket, 'SOCK_CLOEXEC', 0)
+    sock = transport.get_extra_info('socket')
+    if sock is not None and (sock.type & ~nonblock & ~cloexec) == socket.SOCK_STREAM:
```
Comment (foolswood) [inline aioamqp/__init__.py]: Isn't AMQP always over TCP? I don't think I understand what you are checking here?

File: aioamqp/protocol.py [hunk 2]
```diff
@@ -165,7 +165,12 @@ def start_connection(self, host, port, login, password, virtualhost, ssl=False,
         yield from self.open(virtualhost, capabilities='', insist=insist)
 
         # wait for open-ok
-        yield from self.dispatch_frame()
+        frame = yield from self.get_frame()
+        yield from self.dispatch_frame(frame)
+        if frame.frame_type == amqp_constants.TYPE_METHOD and \
+            frame.class_id == amqp_constants.CLASS_CONNECTION and \
```
Comment (foolswood) [inline aioamqp/protocol.py]: Perhaps use the more usual parenthesised and indented continuation, so the indentation of the block and the conditions differ.

Comment (RemiCardona) [inline aioamqp/protocol.py]: fixup coming up

**Comment by foolswood** [comment]
+1 from me

**Comment by RemiCardona** [comment]
@foolswood Thanks for the review

**Pull request merged**